packages/front-end/components/Settings/ApiKeys.tsx

```diff
@@ -31,6 +36,14 @@ const ApiKeys: FC = () => {
     }
   });
 
+  const revealSecret = async (apiKey: string) => {
+    const res = await apiCall<{ secret: string }>(`/key/secret/${apiKey}`, {
+      method: "GET",
+    });
+
+    setSecretKeys({ [apiKey]: res.secret });
```

Comment:
This will override any previously revealed keys.  Is that intentional?

Reply:
I did do this intentionally since these keys can be somewhat confidential. I figured if we were going through the effort to hide them by default, it made sense to only show one at a time.

This does potentially result in multiple API calls, if someone is switching back and forth, but I'm not sure that will really be an issue.